Tests/UnitTests/Examples/Algorithms/Alignment/AlignmentGroupingTests.cpp

```diff
@@ -92,3 +91,26 @@ BOOST_AUTO_TEST_CASE(Multiple_test) {
   BOOST_CHECK(aGroup.has(geoId3));
   BOOST_CHECK(!aGroup.has(geoId4));
 }
+
+BOOST_AUTO_TEST_CASE(CalculateCenterOfMassTest) {
+  ActsExamples::GeometryContext gctx; 
+  std::vector<const Acts::Surface*> surfaces;
+
+
+  surfaces.push_back(new ActsExamples::SurfaceTest(Acts::GeometryIdentifier(), Eigen::Vector3d(1, 2, 3)));
+  surfaces.push_back(new ActsExamples::SurfaceTest(Acts::GeometryIdentifier(), Eigen::Vector3d(-1, -2, -3)));
+  surfaces.push_back(new ActsExamples::SurfaceTest(Acts::GeometryIdentifier(), Eigen::Vector3d(2, 4, 6)));
+
+  // Expected center of mass is the average of the positions
+  Eigen::Vector3d expectedCOM(2.0 / 3, 4.0 / 3, 6.0 / 3);
+
+  // Calculate center of mass
+  Eigen::Vector3d calculatedCOM = ActsExamples::calculateCenterOfMass(surfaces, gctx);
+
+  BOOST_CHECK(calculatedCom.isApprox(expectedCom, 1e-3)); 
+
+  
+  for (auto* surface : surfaces) {
+    delete surface;
+  }
```

Comment:
```suggestion
```
Is there a strong reason, why you do a manual clean-up? At the end of the scope the surfaces will be removed anyway.